app/models/registration.rb

```diff
@@ -386,6 +386,19 @@ def events_to_associated_events(events)
     end
   end
 
+  def permit_user_cancellation?
+    case competition.competitor_can_cancel.to_sym
+    when :always
+      true
+    when :not_accepted
+      !accepted?
+    when :unpaid
+      paid_entry_fees == 0
+    else
+      false
```

Comment:
FWIW this state is never reachable. Rails `enum` introduces validations under the hood that make sure you can never set any (in this case, integer) value that is not covered by the enum.

Do we _want_ cases where the cancellation policy is just "straight-out nope"? In that case, we need a separate `never` enum constant.